x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -212,20 +221,28 @@ void deleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete, TimeValu
         logger.info("starting snapshot retention deletion for [{}] snapshots", count);
         long startTime = nowNanoSupplier.getAsLong();
         int deleted = 0;
+        int failed = 0;
         for (Map.Entry<String, List<SnapshotInfo>> entry : snapshotsToDelete.entrySet()) {
             String repo = entry.getKey();
             List<SnapshotInfo> snapshots = entry.getValue();
             for (SnapshotInfo info : snapshots) {
-                deleteSnapshot(repo, info.snapshotId());
-                deleted++;
+                Optional<SnapshotHistoryItem> result = deleteSnapshot(repo, info);
+                if (result.isPresent()) {
+                    if (result.get().isSuccess()) {
```

Comment:
I think this misses the failed path, where the delete returned a response (so `isPresent()` is true), but the response is a failure. It should increment `failed` in that case?